Bottom: 561b5c8f230b9b1de1257ffeef0c4d63cebee71e
Top:    04f9d58c1850f18484c723d68bfe7370ff70f7de
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-12-11 21:56:15 +0000

Refresh of toy-rendering-attempt

---

diff --git a/src/hid/step/model.cpp b/src/hid/step/model.cpp
index 690519e..efa3a11 100644
--- a/src/hid/step/model.cpp
+++ b/src/hid/step/model.cpp
@@ -1041,8 +1041,8 @@ process_sr_or_subtype(InstMgr *instance_list, SdaiShape_representation *sr, proc
       rz = ((RealNode *)child_axis->ref_direction_ ()->direction_ratios_ ()->GetHead ()->NextNode ()->NextNode ())->value;
 
       /* XXX: Looking only at the target vector.. need to find some examples where the parent transform coordinate system is not unity to get this correct */
-      translate_origin (info->current_transform, ox, oy, oz);
       rotate_basis (info->current_transform, ax, ay, az, rx, ry, rz);
+      translate_origin (info->current_transform, ox, oy, oz);
 
       ox = ((RealNode *)parent_axis->location_ ()->coordinates_ ()->GetHead ())->value;
       oy = ((RealNode *)parent_axis->location_ ()->coordinates_ ()->GetHead ()->NextNode ())->value;
